src/utils.py

```diff
@@ -1,109 +1,301 @@
+        cv2.imwrite(os.path.join(image_dir, image_name), img)
+        # Add to annotations
+        annotations["images"].append(image_info)
+        annotations["annotations"].extend(axon_annotations)
+        annotations["annotations"].extend(myelin_annotations)
+
+def download_default_sem_dataset():
+    if not os.path.exists("data_axondeepseg_sem"):
+        subprocess.run(["git", "clone", SEM_DATASET_URL])
+
+def preprocess_data_yolo(data_dir: str = "data_axondeepseg_sem"):
+    """Preprocesses the loaded BIDS data for object detection.
+
+    Steps:
+    1. Load
+    2. Normalize + windowing
+    3. find regions axons Use skimage.measures.regionprops()
+    4. Croping + Resizing
+
+    LABELS FORMAT DESCRIPTION FOR YOLO: similar to COCO
+    - One *.txt file per image
+    - One row per object in class x_center y_center width height format.
+    - Box coordinates must be in normalized xywh format (from 0 to 1)
+    - Class numbers should be zero-indexed
+    """
+
+    # TODO: Issue of raw path
+    processed_images_dir = "data-yolo/images"
+    processed_masks_dir = "data-yolo/labels"
+
+    train_images_dir = os.path.join(processed_images_dir, "train")
+    train_masks_dir = os.path.join(processed_masks_dir, "train")
+    val_images_dir = os.path.join(processed_images_dir, "val")
+    val_masks_dir = os.path.join(processed_masks_dir, "val")
+    test_images_dir = os.path.join(processed_images_dir, "test")
+    test_masks_dir = os.path.join(processed_masks_dir, "test")
+
+    if data_dir == "data_axondeepseg_sem":
+        download_default_sem_dataset()
+
+    #Creating directories
+    os.makedirs(train_images_dir, exist_ok=True)
+    os.makedirs(train_masks_dir, exist_ok=True)
+    os.makedirs(val_images_dir, exist_ok=True)
+    os.makedirs(val_masks_dir, exist_ok=True)
+    os.makedirs(test_images_dir, exist_ok=True)
+    os.makedirs(test_masks_dir, exist_ok=True)
+
+    data_dict = utils.load_bids_images(data_dir)
+    bbox_data = []
+
+    #Collect images and masks in a format for YOLO
+    image_mask_pairs = []
+
+    for subject in tqdm(data_dict.keys(), desc='Loading dataset for YOLO conversion.'):
+        if subject == "sidecar":
+            continue
+
+        pixel_size = data_dict[subject]['sidecar']
+        for sample in data_dict[subject].keys():
+            if sample == "sidecar":
+                continue
+            img_path = data_dict[subject][sample]['image']
+            axon_seg_path = data_dict[subject][sample]['axon']
+            myelin_seg_path = data_dict[subject][sample]['myelin']
+            image_name = f"{subject}_{sample}.png"
+            label_name = f"{subject}_{sample}.txt"
+
+            # Load and preprocess the images and the segmentation mask
+            img = utils.load_bids_image(img_path, pixel_size)
+            img = utils.normalize_and_window(img)
+
+            # Load segmentation masks and find regions
+            axon_seg = cv2.imread(axon_seg_path, cv2.IMREAD_GRAYSCALE)

```

Comment:
Not a requirement, but in the ads_utils module of AxonDeepSeg we wrote an imread function that handles loading images as grayscale that handles some edge cases, this could alternatively be used via import: https://github.com/axondeepseg/axondeepseg/blob/7b79f486f731dfa05f9fbfab1a24e4710a652c9b/AxonDeepSeg/ads_utils.py#L94